packages/fx-core/src/plugins/resource/appstudio/plugin.ts

```diff
@@ -590,26 +591,35 @@ export class AppStudioPluginImpl {
     return undefined;
   }
 
-  public async buildTeamsAppPackage(ctx: PluginContext): Promise<string> {
+  public async buildTeamsAppPackage(ctx: PluginContext, isLocalDebug: boolean): Promise<string> {
     let manifestString: string | undefined = undefined;
 
     if (!ctx.envInfo?.envName) {
       throw new Error("Failed to get target environment name from plugin context.");
     }
 
     const appDirectory = await getAppDirectory(ctx.root);
-    const zipFileName: string = isMultiEnvEnabled()
-      ? `${ctx.root}/${AppPackageFolderName}/appPackage.${ctx.envInfo.envName}.zip`
-      : `${ctx.root}/${AppPackageFolderName}/appPackage.zip`;
+    let zipFileName: string;
+    if (isMultiEnvEnabled()) {
```

Comment:
So this partially applies to multi-env, app package will be generated when publish/zip/provision, but the manifest file will not be generated?

Reply:
It applies to multi-env, the manifest file will also be generated from line 684

Reply:
Sorry, typo. I mean non multi env. If it's not multi-env,  the app package will still be generated when publish/zip/provison in current logic, right?

Reply:
no, for non multi-env, only zip/publish will trigger build. Provision/local debug won't.